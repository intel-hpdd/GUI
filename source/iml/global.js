// Copyright (c) 2020 DDN. All rights reserved.
// Use of this source code is governed by a MIT-style
// license that can be found in the LICENSE file.

try {
  // Assign directly to window due to https://github.com/rustwasm/wasm-bindgen/pull/1995
  // eslint-disable-next-line no-undef, camelcase
  window.wasm_bindgen = wasm_bindgen;
} catch (e) {
  window.wasm_bindgen = {};
}

export default window;
