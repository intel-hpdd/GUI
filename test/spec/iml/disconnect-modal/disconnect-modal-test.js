// @flow

import { render } from "inferno";
import global from "../../../../source/iml/global.js";
import DisconnectModal from "../../../../source/iml/disconnect-modal/disconnect-modal.js";

describe("disconnect modal", () => {
  let el;

  beforeEach(() => {
    el = document.createElement("div");
    if (global.document.body) global.document.body.appendChild(el);

    render(<DisconnectModal />, el);
  });

  afterEach(() => {
    render(null, el);
    if (global.document.body) global.document.body.removeChild(el);
  });

  it("should generate the modal", () => {
    expect(el).toMatchSnapshot();
  });
});
