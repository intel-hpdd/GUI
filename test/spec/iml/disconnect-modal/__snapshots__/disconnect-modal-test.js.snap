// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`disconnect modal should generate the modal 1`] = `
<div>
  <div
    class="modal fade in disconnect-modal"
    role="dialog"
    style="display: block;"
    tabindex="-1"
  >
    <div
      class="modal-dialog"
      role="document"
    >
      <div
        class="modal-content"
      >
        <div
          class="modal-body "
        >
          <h3>
            Disconnected From Server, Retrying. 
            <i
              class="fa fa-spinner fa-spin fa-lg"
            />
          </h3>
        </div>
      </div>
    </div>
  </div>
  <div
    class="modal-backdrop fade in "
  />
</div>
`;
